pandas/_libs/index.pyx

```diff
@@ -267,6 +271,25 @@ cdef class IndexEngine:
         self._ensure_mapping_populated()
         return self.mapping.lookup(values)
 
+    def get_stargets(self, ndarray targets) -> set:
+        return set(targets)
+
+    def convert_val_if_nan(self, object val) -> object:
+        # unable to utilize val if nan when updating
+        # hashable data structures (ie. sets, dict)
+        if checknull(val):
+            return -1
+        else:
+            return val
+
+    def should_update_d(self, object target, object val) -> bool:
```

Comment:
i think this can be subclass-specific too, only ObjectEngine and Float(64|32)Engine need to check for NAs